xdsl/dialects/riscv.py

```diff
@@ -1584,6 +1586,39 @@ class EcallOp(NullaryOperation):
     name = "riscv.ecall"
 
 
+@irdl_op_definition
+class DirectiveOp(IRDLOperation, RISCVOp):
+    """
+    The directive operation is used to emit assembler directives (e.g. .word; .text; .data; etc.)
+    """
+
+    name = "riscv.directive"
+    directive: OpAttr[StringAttr]
+    value: OptOpAttr[StringAttr]
+    data: OptSingleBlockRegion
+
+    def __init__(
+        self,
+        directive: str | StringAttr,
+        value: str | StringAttr | None,
```

Comment:
I can see some directives [here](https://github.com/riscv-non-isa/riscv-asm-manual/blob/master/riscv-asm.md) that seem to take more than one value.
Should we make the value field variadic?
Might be more suitable for a next PR, what do you think @superlopuh ?

Reply:
The value here is just the string expression, so we'll need to handle converting multiple arguments to a single string during the lowering, but this operation seems ok to me

Reply:
Okay I see, then it's fine! Thanks!